client/state/support/actions.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SUPPORT_USER_ACTIVATED,
+	SUPPORT_USER_DEACTIVATED,
+	SUPPORT_USER_FETCH_TOKEN,
+} from 'state/action-types';
+
+import User from 'lib/user';
+
+import userSettings from 'lib/user-settings';
+
+/**
+ * Requests a support user token, then dispatches the relevant actions upon response
+ * 
+ * @param  {string} supportUser     Support username
+ * @param  {string} supportPassword Support password
+ * @return {thunk}                  The action thunk
+ */
+export function supportUserFetchToken( supportUser, supportPassword ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SUPPORT_USER_FETCH_TOKEN,
+			supportUser
+		} );
+
+		if ( supportUser && supportPassword ) {
+			let user = new User();
```

Comment:
in that case, I would argue that `const` is more appropriate. At a very practical level it doesn't matter because Babel turns them all into `var` at the moment, but thinking about the future I think we should use the constructs based on how they actually work instead of using them in a way that expresses a subtly different behavior.

```js
const o = { a: 1 };
o.a = 5; // no problem
o = { b: 2 }; // big problem
```

The above is appropriate because the JavaScript `const` declaration _only_ concerns the binding of an assignment to a name. In this case, `o` is only ever assigned the reference to memory where the corresponding object is stored. We couldn't, however, reassign it to another memory location or string or anything.

For a similar reason we are free to / should use `const` within loops and closures:

```js
range( 10 ).map( i => {
	const powerOf = partialRight( Math.pow, i );

	return range( 10 ).map( powerOf );
} )
```

Even though we change `powerOf` on each iteration, it's still within its own scope and can benefit from the constraint that `const` provides.

It's confusing to me to read this here as a `let` because it stands out against the language idiom: _why is this not a `const`? am I missing the place it's reassigned/mutated?`

That being said, I understand your point, but feel like it's not a sufficient ground to abandon `const` and all the possible future value we gain from declaring certain bindings as unchangeable.